src/hydra/fetchJsonLd.ts

```diff
@@ -25,22 +25,28 @@ export default async function fetchJsonLd(
   return response.json().then(body => ({ response, body, document: body }));
 }
 
-function setHeaders(options: RequestInit): RequestInit {
-  if (!(options.headers instanceof Headers)) {
-    options.headers = new Headers(options.headers);
+function setHeaders(options: RequestInitExtended): RequestInit {
+  const result = { ...options };
```

Comment:
I think it would be better to typehint `result` here instead of casting it at the end.